--- src/gui/itemviews/qtreeview.cpp~    2008-04-23 22:39:39.000000000 +0200
+++ src/gui/itemviews/qtreeview.cpp     2008-06-19 14:59:24.000000000 +0200
@@ -2598,6 +2598,8 @@
     QStyleOptionViewItemV4 option = d->viewOptionsV4();
     const QVector<QTreeViewItem> viewItems = d->viewItems;
     for (int i = 0; i < viewItems.count(); ++i) {
+        if ( viewItems.at(i).spanning )
+            continue;
         QModelIndex index = viewItems.at(i).index;
         index = index.sibling(index.row(), column);
         QWidget *editor = d->editorForIndex(index);
