Index: admin/acinclude.m4.in
===================================================================
RCS file: /home/kde/kde-common/admin/acinclude.m4.in,v
retrieving revision 2.461
diff -u -r2.461 acinclude.m4.in
--- admin/acinclude.m4.in	26 Jul 2004 14:13:36 -0000	2.461
+++ admin/acinclude.m4.in	1 Aug 2004 21:23:13 -0000
@@ -1665,15 +1665,6 @@
        KDE_USE_CLOSURE_TRUE="#"
        KDE_USE_CLOSURE_FALSE=""
        KDE_NO_UNDEFINED=""
-       case $host in 
-         *-*-linux-gnu)
-           KDE_CHECK_COMPILER_FLAG([Wl,--no-undefined],
-                [KDE_CHECK_COMPILER_FLAG([Wl,--allow-shlib-undefined],
-		[KDE_NO_UNDEFINED="-Wl,--no-undefined -Wl,--allow-shlib-undefined"],
-		[KDE_NO_UNDEFINED=""])],
-	    [KDE_NO_UNDEFINED=""])
-           ;;
-       esac
   fi
   AC_SUBST(KDE_USE_CLOSURE_TRUE)
   AC_SUBST(KDE_USE_CLOSURE_FALSE)
