Index: admin/detect-autoconf.sh
===================================================================
RCS file: /home/kde/kde-common/admin/detect-autoconf.sh,v
retrieving revision 1.13
diff -u -r1.13 detect-autoconf.sh
--- admin/detect-autoconf.sh	24 May 2004 11:36:58 -0000	1.13
+++ admin/detect-autoconf.sh	6 Nov 2004 20:05:20 -0000
@@ -10,8 +10,8 @@
 # Please add higher versions first. The last version number is the minimum
 # needed to compile KDE. Do not forget to include the name/version #
 # separator if one is present, e.g. -1.2 where - is the separator.
-KDE_AUTOCONF_VERS="-2.58 -2.57 257 -2.54 -2.53a -2.53 -2.52 -2.5x"
-KDE_AUTOMAKE_VERS="-1.7 17 -1.6"
+KDE_AUTOCONF_VERS="-2.58 -2.57 257 -2.54 -2.53a -2.53 -2.52 -2.5x 2.50"
+KDE_AUTOMAKE_VERS="-1.9 -1.8 -1.7 17 -1.6"
 
 # We don't use variable here for remembering the type ... strings. Local 
 # variables are not that portable, but we fear namespace issues with our
@@ -22,7 +22,9 @@
     if test -x "`$WHICH $AUTOCONF$kde_autoconf_version 2>/dev/null`"; then
       AUTOCONF="`$WHICH $AUTOCONF$kde_autoconf_version`"
       AUTOHEADER="`$WHICH $AUTOHEADER$kde_autoconf_version`"
-      AUTOM4TE="`$WHICH $AUTOM4TE$kde_autoconf_version`"
+      if test -x "`$WHICH $AUTOM4TE$kde_autoconf_version 2>/dev/null`"; then
+	AUTOM4TE="`$WHICH $AUTOM4TE$kde_autoconf_version`"
+      fi
       break
   fi
   done
