Index: korganizer/Makefile.am
===================================================================
RCS file: /home/kde/kdepim/korganizer/Makefile.am,v
retrieving revision 1.275
diff -u -r1.275 Makefile.am
--- korganizer/Makefile.am	8 Jul 2004 13:13:01 -0000	1.275
+++ korganizer/Makefile.am	5 Aug 2004 14:47:29 -0000
@@ -48,8 +48,8 @@
 timezone_SOURCES = timezone.cpp
 timezone_COMPILE_FIRST = koprefs_base.h
 
-lib_LTLIBRARIES = libkorganizer.la libkorganizer_eventviewer.la \
-                  libkorganizer_calendar.la
+lib_LTLIBRARIES = libkorganizer_eventviewer.la \
+                  libkorganizer_calendar.la libkorganizer.la
 
 libkorganizer_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -version-info 1:0
 libkorganizer_la_LIBADD  = $(LIB_KPARTS) $(LIB_KFILE) \
