Index: configure.in.in
===================================================================
RCS file: /home/kde/kdepim/configure.in.in,v
retrieving revision 1.35
diff -u -r1.35 configure.in.in
--- configure.in.in	10 Jun 2004 12:36:15 -0000	1.35
+++ configure.in.in	8 Aug 2004 20:42:41 -0000
@@ -1,4 +1,6 @@
 #MIN_CONFIG(3.3)
+AM_INIT_AUTOMAKE(kdepim, @VERSION@)
+
 AC_HEADER_DIRENT
 AC_HEADER_STDC
 AC_CHECK_HEADERS(fcntl.h sys/sysctl.h sys/param.h sys/time.h sys/types.h unistd.h)
