Index: tools/gpg-connect-agent.c
===================================================================
--- tools/gpg-connect-agent.c	(revision 4589)
+++ tools/gpg-connect-agent.c	(working copy)
@@ -26,6 +26,13 @@
 #include <ctype.h>
 #include <assuan.h>
 
+#ifdef _WIN32
+# include <direct.h>
+# define getcwd _getcwd
+#else
+# include <unistd.h>
+#endif
+
 #include "i18n.h"
 #include "../common/util.h"
 #include "../common/asshelp.h"
@@ -205,10 +212,13 @@
 
   for (var = variable_table; var; var = var->next)
     if (!strcmp (var->name, name))
-      break;
-  if (!var || !var->value)
-    return NULL;
-  return var->value;
+      return var->value;
+  if(!strcmp (name, "cwd"))
+    {
+      static char cwdbuf[256];
+      return getcwd( cwdbuf, sizeof cwdbuf );
+    }
+  return getenv (name);
 }
 
 
